rasa/core/brokers/kafka.py

```diff
@@ -124,67 +124,73 @@ def publish(
                     f"Could not publish message to kafka url '{self.url}'. "
                     f"Failed with error: {e}"
                 )
-                connected = self.producer.bootstrap_connected()
-                if not connected:
-                    self._close()
+                try:
```

Comment:
the recommendation for `BufferError` is to call `poll`: https://github.com/confluentinc/confluent-kafka-python/blob/5936d0b465ad45eaff9c02ed7829f02d7c797cd7/src/confluent_kafka/serializing_producer.py#L117
not sure what to do about generic `KafkaException` - maybe best to raise